src/service/navigation.js

```diff
@@ -528,14 +545,10 @@ export class Navigation {
    * @param {!Event} e
    * @param {!Element} target
    * @param {!Location} tgtLoc
+   * @param {!Location} curLoc
    * @private
    */
-  handleNavClick_(e, target, tgtLoc) {
-    // In test mode, we're not able to properly fix the anchor tag's base URL.
-    // So, we have to use the (mocked) window's location instead.
-    const baseHref =
-      getMode().test && !this.isEmbed_ ? this.ampdoc.win.location.href : '';
-    const curLoc = this.parseUrl_(baseHref);
+  handleNavClick_(e, target, tgtLoc, curLoc) {
     const tgtHref = `${tgtLoc.origin}${tgtLoc.pathname}${tgtLoc.search}`;
     const curHref = `${curLoc.origin}${curLoc.pathname}${curLoc.search}`;
```

Comment:
Here it is. It seems we have the same code duplicated in `handleNavClick_` and `handleClick_` now, can we clean this up?

Reply:
For sure. I introduced a `getHref()` helper function. 
I'm a bit hesitated to add the function to url.js because it requires the input to be `{!Location}`. So I added it to this file. Let me know if you're good with this. Thank

Reply:
So I don't understand what `handleClick_`'s responsibility is anymore. It seems like both methods try to handle internal clicks, but `handleNavClick_` can't actually receive an internal click anymore (because it's called by `handleClick_` after `handleClick_` has handled an internal click).

Reply:
True, I had that same feeling. My original thought was to divide `handleNavClick_` to two methods `handleInternalClick_()` and `handleExternalClick_()`. 
But in theory it is possible that the anchorMutator changes the href from an external url back to an internal url. And I feel it's safer to parse the location again after the mutations. 
However I also don't think that's going to be the case now. So I'm also fine with assuming that the target href is still external url after the mutations. Let me know what do you think is the best. Thanks.

Reply:
As agreed offline, change the code to check href to apply mutator. PTAL